### PR TITLE
cleanup(generator): separate golden from hand-crafted files

### PR DESCRIPTION
**Pull request by coryan**
Move the hand-crafted files out of the
`generator/integration_tests/golden/` subdirectory, including CMake and Bazel support files.  I think this will make it easier to work in these files, as they will format and work as regular files.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10085)

### CODE REVIEW

File: generator/integration_tests/BUILD.bazel [hunk 1]
```diff
@@ -59,6 +59,70 @@ cc_grpc_library(
...
+        "golden/internal/*.cc",
+    ]),
+)
+
+filegroup(
+    name = "golden_hdrs",
+    srcs = glob([
+        "golden/*.h",
+        "golden/internal/*.h",
+        "golden/mocks/*.h",
```
Comment (dbolduc) [inline generator/integration_tests/BUILD.bazel:78]: should these be split out into their own `google_cloud_cpp_generator_golden_mocks` library?

File: generator/integration_tests/BUILD.bazel [hunk 2]
```diff
@@ -59,6 +59,70 @@ cc_grpc_library(
     name = "google_cloud_cpp_generator_testing_cc_grpc",
     srcs = [":google_cloud_cpp_generator_testing_proto"],
     grpc_only = True,
-    visibility = [":__subpackages__"],
     deps = [":google_cloud_cpp_generator_testing_cc_proto"],
 )
+
+filegroup(
```
Comment (dbolduc) [inline generator/integration_tests/BUILD.bazel:65]: TIL. `filegroup`. It looks tidier than what we currently do with: `SOURCE_GLOB = "**/*.cc"` in our generated `BUILD.bazel`s. I do not think it is worth updating though.

File: generator/integration_tests/CMakeLists.txt [hunk 3]
```diff
@@ -44,12 +51,102 @@ target_link_libraries(
...
-# to build all the tests everytime they create a new package or when the package
-# is installed from source.
-if (BUILD_TESTING)
-    add_subdirectory(golden)
-endif (BUILD_TESTING)
+file(
+    GLOB google_cloud_cpp_generator_golden_files
+    RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
+    "golden/*.h" "golden/*.cc" "golden/internal/*.h" "golden/internal/*.cc"
+    "golden/mocks/*.h")
```
Comment (dbolduc) [inline generator/integration_tests/CMakeLists.txt:61]: same comment about splitting out the mocks. it's fine if you don't think this is worth doing.

File: generator/integration_tests/CMakeLists.txt [hunk 4]
```diff
@@ -44,12 +51,102 @@ target_link_libraries(
...
+    tests/golden_thing_admin_stub_test.cc
+    tests/plumbing_test.cc)
+
+# Export the list of unit tests to a .bzl file so we do not need to maintain the
+# list in two places.
+export_list_to_bazel("google_cloud_cpp_generator_golden_tests.bzl"
+                     "google_cloud_cpp_generator_golden_tests" YEAR "2020")
+
+# Create a custom target so we can say "build all the tests"
+add_custom_target(generator-golden-tests)
```
Comment (dbolduc) [inline generator/integration_tests/CMakeLists.txt:127]: hm do we do this anywhere else? is it necessary?

**Review by dbolduc — APPROVED** [review]

File: generator/integration_tests/CMakeLists.txt (hunk 4 above)
Comment (coryan) [reply generator/integration_tests/CMakeLists.txt]: Removed.

File: generator/integration_tests/BUILD.bazel (hunk 2 above)
Comment (coryan) [reply generator/integration_tests/BUILD.bazel:65]: Ack

File: generator/integration_tests/BUILD.bazel (hunk 1 above)
Comment (coryan) [reply generator/integration_tests/BUILD.bazel:78]: I will leave these as-is until I find a reason to split them.

File: generator/integration_tests/CMakeLists.txt (hunk 3 above)
Comment (coryan) [reply generator/integration_tests/CMakeLists.txt:61]: Ack

**Pull request merged**